### PR TITLE
feat: better logging settings, log client in certain logs

### PR DESCRIPTION
**Pull request by brianluong**
# Description

* Logging client in certain logs is very helpful 
* Make the log formats between production/development mode the same. Single format allows for consistent ingestion on the Datadog side. Also format log timestamps as human readable timestamps  instead of epoch millis. 

## Type of change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] 😎 New feature (non-breaking change which adds functionality)
- [ ] ⁉️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] ⚒️ Refactor (no functional changes)
- [ ] 📖 Documentation (updating or adding docs)

# How Has This Been Tested?

`-e ENV=production -e LOG_LEVEL=debug` works

### CODE REVIEW

File: cmd/gateway/main.go [hunk 1]
```diff
@@ -94,9 +95,15 @@ func main() {
...
 func setupGlobalLogger(env string) (logger *zap.Logger, err error) {
 	if env == "production" {
-		logger, err = zap.NewProduction()
+		zapConfig := zap.NewProductionConfig()
+		zapConfig.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+		logger, err = zapConfig.Build()
 	} else {
-		logger, err = zap.NewDevelopment()
+		zapConfig := zap.NewDevelopmentConfig()
+		zapConfig.Encoding = "json"
```
Comment (dianwen) [inline cmd/gateway/main.go:103]: I find JSON in development really hard to read, which makes developing slower.

Taking a step back, what's the use case for sending dev logs to datadog?

**Review by dianwen — COMMENTED** [review]
Client logging changes look good! Left a comment about the dev logging config.

File: cmd/gateway/main.go (hunk 1 above)
Comment (brianluong) [reply cmd/gateway/main.go:103]: Logging is only controlled by the development/production flags atm. So to get debug logs we have to turn on development. We might want to do this for debugging issues.

Zap's baked in development config has more than just debug logs. For example it includes stack traces for anything above warn. 

The better solution perhaps is to expose a parameter called log level, and pipe that into the zap logger's config.

Comment (dianwen) [reply cmd/gateway/main.go:103]: Gotcha.

Yeah, exposing log level as an env variable makes a lot of sense. No need to always couple the log level to the logging environment.

File: cmd/gateway/main.go [hunk 2]
```diff
@@ -92,11 +90,34 @@ func main() {
 	}
 }
 
-func setupGlobalLogger(env string) (logger *zap.Logger, err error) {
-	if env == "production" {
-		logger, err = zap.NewProduction()
+func setupGlobalLogger(env, logLevel string) (logger *zap.Logger, err error) {
+	if env == "" || env == "production" {
+		if logLevel == "" {
+			logLevel = "info"
```
Comment (dianwen) [inline cmd/gateway/main.go:96]: Consider extracting into a `defaultProductionLogLevel` constant.

Comment (dianwen) [reply cmd/gateway/main.go:96]: Same w/ development

**Review by dianwen — APPROVED** [review]
🚢 :shipit:!

**Pull request merged**